Added FAQ

Zxun2 opened the pull request:
The issue can be found [here](https://github.com/NUS-Fintech-Society/SD_Fintech_Website/issues/99).

Laptop view: 
![image](https://user-images.githubusercontent.com/63457492/151646889-994cddaf-b0fb-4ca3-b616-119aeb878115.png)

Mobile view: 
![image](https://user-images.githubusercontent.com/63457492/151646903-85354286-da41-44ac-b4de-d6e4619a7590.png)

Let me know if there's anything I need to change!

Zxun2 committed:
Clean up

Zxun2 committed:
Fixed build error

zzhikai commented on src/components/FAQ/index.jsx:
```diff
@@ -0,0 +1,148 @@
+      margin: '1rem 0 0 0',
+    },
+    '& .MuiAccordionSummary-content': {
+      paddng: '0 0.8rem 0 0.8rem',
```
typo here

zzhikai replied:
```suggestion
      padding: '0 0.8rem 0 0.8rem',
```
Typo here, fixing it will shift the content to the right and need to align with Details-root
![Screenshot 2022-01-29 at 9 38 55 PM](https://user-images.githubusercontent.com/77470406/151663213-d7f9d4f9-920f-4cea-805a-bcb65dc9be4a.png)

zzhikai commented on src/components/FAQ/index.jsx:
```diff
@@ -0,0 +1,148 @@
+      margin: 0,
+    },
+    '& .MuiAccordionDetails-root': {
+      padding: '1rem 1rem 0 1rem',
```
```suggestion
      padding: '1rem 1.8rem 0 1.8rem',
```

I guess can adjust to the question alignment, seems more neat with more padding around
![Screenshot 2022-01-29 at 9 46 58 PM](https://user-images.githubusercontent.com/77470406/151663501-6dc8dbde-f992-4657-a25c-85ef0ff680aa.png)

Zxun2 replied:
Okay!

zzhikai reviewed (changes requested):
Just need to fix the typo and adjust accordingly.
Not sure if this was intended in the design, but implementation here is slightly different from FTS-month, opening one FAQ closes the one that is opened before. Will need to clarify with designers.

Zxun2 committed:
Fix padding

Zxun2 commented:
Alright, fix the issue with the padding. Let me know if we prefer the Accordion on the FTS-month website instead!

Zxun2 committed:
Change accordion behaviour

Zxun2 commented:
Added the new behavior! Do see if it's good now!

jerrenss commented on src/components/FAQ/index.jsx:
```diff
@@ -0,0 +1,154 @@
+                  </AccordionSummary>
+                  <AccordionDetails>
+                    {faq.A.split('\n').map((para, idx) => {
+                      const strToBold = 'nusfintech.ops@gmail.com'
```
Hey @Zxun2 this current implementation will work for most of the simple FAQs. However, in the event where there are requests to bold multiple words or include line breaks, this might not be sufficient.

I was thinking we can try experimenting usage of a HTML to text approach which might be more scalable? Our answers will be written in HTML eg. `<p>This is the <h3>answer</h3></p>`, which will then be rendered as styled strings. This might be achievable with dangerouslySetInnerHTML or other npm libraries.

https://www.npmjs.com/package/html-react-parser
https://www.npmjs.com/package/html-to-text
https://dev.to/jobpick/how-to-render-html-string-in-a-react-component-3kd2

What do you think?

Zxun2 replied:
Interesting, I'll give it a try! Some things I'd like to clarify. 

Would inline-html (for the respective sections) look something like this?

For questions:  `<p> Questions that I would like to ask <p>` or would it be any of the heading tags? `<h1><h1/>` etc
Edit: I don't think it matters...I'll replace all tags in the heading with MUI Typography

For answers:  `<p>Hello <b>World<b></p><p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim.<p>` In this case, the line break is for each `<p></p>` tag.

Also, would there be any inline attributes in the raw HTML data that I need to account for? `<p id="main" class="main">Hello World</p>`

Thank you!

Zxun2 committed:
Added HTML-to-Text

Zxun2 commented:
Added the functionality! 

Some changes: 
1. Convert FAQs.js to FAQs.json.
2. Added HTML-to-Text
3. Added some padding to details-root 

I really like this approach!! Do let me know if there's anything else to change ^ __ ^

jerrenss commented on src/data/FAQs.json:
```diff
@@ -0,0 +1,56 @@
+  "qa": [
+    {
+      "id": 1,
+      "section": "<h1>Recruitment</h1>",
```
Nice attempt! Since our section headers and questions are kinda fixed in terms of styling, let's just use string for them, eg.` "section": "Recruitment"` and `"Q": "Who do we reach out to if we have more enquiries?"`. So we don't need to do any html to string conversion for sections and questions, **but only for answers**

jerrenss replied:
Also noticed you added an id in `<p id='main'>` of each answer, not sure if that is necessary. If it isn't, can remove too.

jerrenss reviewed (commented):
Oh lastly, can remove package-lock.json and add additional paddingTop and paddingBottom to each card, 8px will do. Thanks!

![image](https://user-images.githubusercontent.com/50544486/151692437-a932e7da-42d6-4d60-a34a-31654ad6691b.png)

Zxun2 committed:
Remove HTT from Sections/Questions

Zxun2 committed:
removed package-lock.json

jerrenss reviewed (approved):
Nice! @zzhikai can help to double check and merge

zzhikai reviewed (approved):
LGTM!
